### ISSUE TITLE
possible 'order' bug

### ISSUE THREAD

**Issue by cvoeten** [cvoeten/buildmer#3]
Thank you for your great package. I want to keep fixed effects  `A*B`, but I get the error. 
> Error in `[<-.data.frame`(`*tmp*`, , p$crit.name, value = c(NA, NA, -4.87465714865097,  : 
  replacement has 5 rows, data has 9

This is the code (actually, I don't really know how to use the 'include' argument.) However, if the 'order' is deleted, it works.

```
fe <- DV~ 1 + A*B
m <- buildmer(f,data=exp3,direction=c('order','backward'),
              control=lmerControl(optimizer='bobyqa'),
              include = fe)
```

_Originally posted by @phliulei in https://github.com/cvoeten/buildmer/issues/1#issuecomment-633676476_

**Comment by phliulei** [comment]
Sure, it's my pleasure, but can you give me your E-mail?

**Comment by cvoeten** [comment]
It's my username at gmail - but if you put up your (possibly anonymized) data here---and give me permission---I can include it in the package as a regression test for future versions.

**Comment by phliulei** [comment]
OK, this is the anonymous data (and my code), and you can use it to do the regression test.
[orderBug.zip](https://github.com/cvoeten/buildmer/files/4678614/orderBug.zip)

**Comment by cvoeten** [comment]
Thanks! The issue turned out to be in `backward()`, which did not work properly with `include`. Can you pull the latest code from GitHub and see if it fixes the problem for your complete dataset as well? If this fixes your problem, I'll release it as buildmer 1.6 --- luckily, that was due for release anyway.

**Comment by phliulei** [comment]
Great! It works. Thank you. 

I have another question. Does this package support ordinal mixed regression, such as `clmm()` in the ordinal package? Or would you like to support the `ordinal::clmm()` package? Because I found few packages fit for ordinal mixed regression.

**Comment by cvoeten** [comment]
The general answer to this type of question is: 'yes, you can use buildmer with any fitting engine you like via `buildcustom()`', but in this case I've tried to take a first stab at a `buildclmm()` function myself. Can you test if that does what you want?

**Comment by phliulei** [comment]
Thank you, but

`Error in mkCrit(crit) ` not find mkCrit function

**Comment by cvoeten** [comment]
It seems like you `source()`d the GitHub version of buildmer, rather than installing it. The correct procedure is to use `remotes::install_github('cvoeten/buildmer')` from a clean R session. You can then access `buildclmm` by using `buildmer:::buildclmm` (yes, those are three colons, as buildclmm is currently not exported).

**Comment by phliulei** [comment]
It does work. Thank you for your patience.

**Comment by cvoeten** [comment]
I've made some further tweaks to the buildclmm code and released buildmer 1.6 with those tweaks on CRAN. Thanks for your help.
